frontend/src/shared/components/tasque-task-editing/task-editing.component.ts

```diff
@@ -0,0 +1,324 @@
+    },
+    {
+      id: 3,
+      name: 'High',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+  ];
+  @Input() public taskStates: TaskState[] = [
+    {
+      id: 1,
+      name: 'To Do',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 2,
+      name: 'In Progress',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 3,
+      name: 'Done',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 4,
+      name: 'Canceled',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+  ];
+
+  @Input() public projects: ProjectModel[] = [
+    {
+      id: 1,
+      name: 'project 1',
+      authorId: 0,
+      organizationId: 0,
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 2,
+      name: 'project 2',
+      authorId: 0,
+      organizationId: 0,
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 3,
+      name: 'project 3',
+      authorId: 0,
+      organizationId: 0,
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+  ];
+
+  @Input() public sprints: SprintModel[] = [
+    {
+      id: 1,
+      name: 'spr1',
+      description: 'sprint desc',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 2,
+      name: 'spr2',
+      description: 'sprint desc',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      id: 3,
+      name: 'spr3',
+      description: 'sprint desc',
+      createdAt: new Date(),
+      updatedAt: new Date()
+    }
+  ];
+  @Input() public users: UserModel[] = [
+    {
+      id: 1,
+      name: 'User1Name',
+      email: 'email'
+    },
+    {
+      id: 2,
+      name: 'User2Name',
+      email: 'email'
+    },
+    {
+      id: 3,
+      name: 'User3Name',
+      email: 'email'
+    }
+  ];
+
+  public editorConfig: AngularEditorConfig = EditorConfig;
+
+  constructor(private sanitizer: DomSanitizer) {
+    super();
+
+    this.task = {
+      id: 1,
+      name: 'Hi',
+      summary: 'Maaaaan',
+      description: 'qweqwe',
+      taskState: {
+        id: 1,
+        name: 'Hi',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      },
+      taskType: {
+        id: 1,
+        name: 'Hi',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      },
+      taskPriority: {
+        id: 1,
+        name: 'Hi',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      },
+      authorId: 1,
+      projectId: 2,
+      sprintId: 3,
+      lastUpdatedById: 4,
+      parentTaskId: 5,
+      createdAt: new Date(),
+      updatedAt: new Date(),
+      deadLine: new Date()
+    };
+    this.currentUser = this.users[1];
+  }
+
+  ngOnInit(): void {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    this.taskProject = this.projects.find((x) => x.id === this.task.projectId)!;
```

Comment:
I think all of these filterings should be done in services

Reply:
Yes, these filterings will be done in services in the next PR that relates more to backend part. But this is just frontent PR.